docs/design/rubygems.md

```diff
@@ -0,0 +1,450 @@
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except BUNLDE_APP_CONFIG),
+we need to set the Bundler configuration options to make the build work.
+We can easily set the environment variables if the local configuration file does not exist.
+
+#### Relevant environment variables
+
+```txt
+BUNDLE_FORCE_RUBY_PLATFORM=true
+BUNDLE_DEPLOYMENT=true
+BUNDLE_CACHE_PATH=${output_dir}/deps/rubygems
+```
+
+**BUNDLE_CACHE_PATH**: The directory that Bundler will place cached gems in when running bundle package,
+and that Bundler will look in when installing gems. Defaults to `vendor/cache`.
+
+**BUNDLE_DEPLOYMENT**: Disallow changes to the Gemfile.
+When the Gemfile is changed, and the lockfile has not been updated, running Bundler commands will be blocked.
+
+**BUNDLE_FORCE_RUBY_PLATFORM**: Ignore the current machine's platform and install only ruby platform gems.
+As a result, gems with native extensions will be compiled from source.
+
+See bundle config [documentation](https://bundler.io/v2.5/man/bundle-config.1.html).
+
+---
+
+Note: _BUNDLE_FORCE_RUBY_PLATFORM_ check is done via gemlock-parser in cachito.
+
+_BUNDLE_DEPLOYMENT_ might be helpful when building an image.
+Using the `--local` flag with the `bundle install`, ensures that all dependencies are installed from the cache without accessing the internet.
+This is uncommon, so we would have to force users to use this flag.
+By setting the `BUNDLE_DEPLOYMENT` environment variable, users do not have use the `--local` flag.
+
+_When installing gems, Bundler will also "fetch" the gems from the cache and store them inside `vendor` directory._
+
+Here is more verbose explanation of the `--deployment` available in the `bundle install` command:
+
+```bash
+--deployment
+```
+
+In deployment mode, Bundler will 'roll-out' the bundle for production or CI use. Please check carefully if you want to have this option enabled in your development environment.
+This option is deprecated in favor of the deployment setting.
+
+There is also `--prefer-local` flag, which will prefer the local cache over the remote source, but it ain't working at all.
+Environment variable `BUNDLE_ALLOW_OFFLINE_INSTALL` is not working either with `bundle install` for some reason,
+which could be probably the most logical solution in this case.
+
+##### Copy
+
+Copy the local configuration file from the user repository to {output_dir} and set BUNDLE_APP_CONFIG to the new location.
+Then, just append all the environment variables needed to the "new" copy of the user configuration file.
+Bundler will rewrite previous values with the new ones when installing gems.
+
+#### Inject
+
+The other solution would be to inject the config file directly and rewrite the values.
+
+### Metadata
```

Comment:
I think we should clarify here how each field of the SBOM component will be resolved from the metadata. My initial ideas:

- name: from .gemspec, or if missing, from the repo url
- version: from .gemspec, or if missing, generate a pseud-version (or leave it empty?)
- purl: main url based on the repo origin (ignore "source_code_uri" from .gemspec?)

Path dependencies should have their purls based on the source repo origin URL, and with the relative path appended.

Reply:
Yeah, but we don't really need to do anything with the `.gemspec` file. _Gemfile.lock_ and repo URL should give us enough information.